File: src/dataset/go/luhn_Exercism_go_thinkverse/claude/ClaudeSonnet4_luhn_v1.go
package luhn

// Valid check is a string is a valid LUHN number.
//
// Benchmarks done with Core i5 760 2.80GHz
//
// Operations  Time (ns/op)
// 271239      3730
//
// 279519      3726
//
// 1189321     1021
//
// 3331646     350
//
// 10332144    112
//
// Thank you to @thomas-holmes[^1], looking at that iteration made my next iteration even faster.
//
// [^1]: https://exercism.org/tracks/go/exercises/luhn/solutions/thomas-holmes
func Valid(id string) bool {
	var sum int
	var digitCount int
	var isSecond bool

	// Process string from right to left
	for i := len(id) - 1; i >= 0; i-- {
		char := id[i]
		
		if char == ' ' {
			continue
		}

		if char < '0' || char > '9' {
			return false
		}

		digit := int(char - '0')
		digitCount++

		if isSecond {
			digit *= 2
			if digit > 9 {
				digit -= 9
			}
		}

		sum += digit
		isSecond = !isSecond
	}

	return digitCount > 1 && sum%10 == 0
}

File: src/dataset/typescript/luhn_Exercism-typescript-chriswilding/claude/ClaudeSonnet4_luhn_v1.ts
const SPACE = 32
const ZERO = 48
const NINE = 57

export default class Luhn {
    public static valid(input: string): boolean {
        let sum = 0
        let digitCount = 0
        let isEven = true

        for (let i = input.length - 1; i >= 0; i--) {
            const c = input.charCodeAt(i)
            
            if (c === SPACE) continue
            if (c < ZERO || c > NINE) return false
            
            let digit = c - ZERO
            
            if (!isEven) {
                digit <<= 1
                if (digit > 9) digit -= 9
            }
            
            sum += digit
            digitCount++
            isEven = !isEven
        }

        return digitCount >= 2 && sum % 10 === 0
    }
}